internal/controller/promotion/argocd.go

```diff
@@ -309,6 +298,31 @@ func (a *argoCDMechanism) mustPerformUpdate(
 		return "", true, nil
 	}
 
+	// Deal with the possibility that the operation was not initiated for the
+	// current freight collection. i.e. Not related to the current promotion.
+	var correctFreightColIDFound bool
+	for _, info := range status.Operation.Info {
+		if info.Name == freightCollectionInfoKey {
+			correctFreightColIDFound = info.Value == freightCol.ID
```

Comment:
Should this consider that _hypothetically_, a Freight collection could somehow end up without an ID?

Reply:
It would be a nice thing to safeguard against, but I think it's a rabbit hole unto itself.

The ID is updated whenever you push to the collection. But because the collection is an API type, all its fields are exported, else marshaling/unmarshaling doesn't work. That introduces two things that I don't love: 1. You can tamper with the underlying collection without the ID being updated. 2. You can mutate the ID in place so it doesn't match the underlying collection anymore. These would be the only two ways that your hypothetical could occur, and except in test cases where we are engineering various scenarios, we never do anything of the sort.

I definitely do not love the _potential_ for a FreightCollection to be mishandled, but I don't feel we have the luxury of sorting that out at the moment. wdyt?